### DISCUSSION TITLE
Is there a limit to redirects and rewrites?

### DISCUSSION THREAD

**Discussion by andrewfairlie** [vercel/next.js#16191] [Help] [answered]
Vercel has a limit in their `vercel.json` of 1024 redirects, but I wonder if Next 9.5's new redirects in `next.config.js` have that same limit, and if not if any limit is applied once on Vercel?

PS: I have no real-world use case for this, I'm just thinking ahead for if that time ever comes.

**Comment by ijjk** [comment, marked as answer]
Hi, that is correct the limit for redirects/rewrites/headers in `next.config.js` is the same as for `vercel.json` when deploying on Vercel

**Reply by andrewfairlie** [reply]
Thank you 👍
